### PR TITLE
Expands contributor guidelines

### PR DESCRIPTION
**Pull request by glwagner**
This PR adds a few sections to `CONTRIBUTING.md` with guidelines for opening an issue, and for becoming an Oceananigans collaborator.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,17 +2,49 @@
...
-practices. New contributors should make sure to read that guide.
+practices. We ask that new contributors read that guide before submitting a pull request.
+
+## Creating issues
+
+The simplest way to contribute to Oceananigans is to create or comment on issues and discussions.
+
+The most useful bug reports:
+
+* Provide an explicit code snippet --- not just a link --- that reproduces the bug in the latest tagged version of Oceananigans. This is sometimes called the ["minimal working example"](https://en.wikipedia.org/wiki/Minimal_working_example#:~:text=In%20computing%2C%20a%20minimal%20working,to%20be%20demonstrated%20and%20reproduced.&text=A%20minimal%20working%20example%20may,short%20self%2Dcontained%20correct%20example.). Reducing bug-producing code to a minimal example can dramatically decrease the time it takes to resolve an issue.
```
Comment (ali-ramadhan) [inline CONTRIBUTING.md:23]:
```suggestion
* Provide an explicit code snippet --- not just a link --- that reproduces the bug in the latest tagged version of Oceananigans. This is sometimes called the ["minimal working example"](https://en.wikipedia.org/wiki/Minimal_working_example). Reducing bug-producing code to a minimal example can dramatically decrease the time it takes to resolve an issue.
```

**Review by ali-ramadhan — APPROVED** [review]
Looks great!

**Comment by navidcy** [comment]
We need to updated the `docs/contributing.md` as well!

**Comment by glwagner** [comment]
Ah, didn't realize those were different. I'll check it out.

**Comment by navidcy** [comment]
I can do that ;)

**Comment by navidcy** [comment]
have a look @glwagner and modify to your liking :)

**Review by navidcy — APPROVED** [review]

**Comment by glwagner** [comment]
Wait, is `docs/contributing.md` just a copy of `/CONTRIBUTING.md`?

**Comment by navidcy** [comment]
Not really, it's an expanded version...

Have a look here: https://clima.github.io/OceananigansDocumentation/previews/PR1765/contributing/

**Comment by glwagner** [comment]
Ahh. I understand (I thought maybe the docs version was specific to contributing to docs). It looks great, I'll merge.

**Pull request merged**